test/e2e/all_test.go

```diff
@@ -0,0 +1,94 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import "testing"
+
+// TestAll is the entrypoint for `make test-e2e` unless you override
+// with: make TEST=Test<foo> test-e2e.
+//
+// The overriding goal of this test is to run as many tests in
+// parallel before running those tests that must run serially. There
+// are two goals: 1) cut down on test execution time and 2) provide
+// explicit ordering for tests that do not expect a rolling update of
+// ingresscontroller pods because a previous test modified the
+// ingressconfig object and the defer logic for cleanup is still
+// runnng when the new test starts.
+func TestAll(t *testing.T) {
+	// This call to Run() will not return until all of its
+	// parallel subtests complete. Each "parallel" test must
+	// invoke t.Parallel().
```

Comment:
Nothing verifies this, right?  As a follow-up, we could add some logic in `hack/verify-e2e-test-all-presence.sh` to verify that all the parallel tests actually call `t.Parallel()`.  (Easier said than done.)